src/main/java/io/icker/factions/ui/InfoGui.java

```diff
@@ -0,0 +1,247 @@
+     */
+    public InfoGui(ServerPlayerEntity player, Faction faction, @Nullable Runnable closeCallback) {
+        super(ScreenHandlerType.GENERIC_9X1, player, false);
+        this.closeCallback = closeCallback;
+
+        MinecraftServer server = player.getServer();
+        UserCache cache = server.getUserCache();
+
+        User user = User.get(player.getUuid());
+        boolean isMember = faction == user.getFaction();
+        List<User> members = faction.getUsers();
+
+        String owner = members.stream().filter(u -> u.rank == User.Rank.OWNER)
+                .map(u -> cache.getByUuid(u.getID())
+                        .orElse(new GameProfile(Util.NIL_UUID, "{Uncached Player}"))
+                        .getName()
+                )
+                .collect(Collectors.joining(", "));
+
+        this.setTitle(Text.literal("Faction info"));
+
+        for (int i = 0; i < 9; i++)
+            this.setSlot(i, new GuiElementBuilder(Items.WHITE_STAINED_GLASS_PANE).hideTooltip());
+
+        // Faction info
+        this.setSlot(0, new GuiElementBuilder(Items.PLAYER_HEAD)
+                .setSkullOwner(isMember ? Icons.GUI_CASTLE_NORMAL : Icons.GUI_CASTLE_OPEN)
+                .setName(Text.literal(faction.getColor() + faction.getName()))
+                .setLore(List.of(
+                                Text.literal(faction.getDescription())
+                                        .setStyle(
+                                                Style.EMPTY
+                                                        .withItalic(false)
+                                                        .withColor(Formatting.WHITE)
+                                        ),
+                                Text.literal("Owner: ")
+                                        .setStyle(
+                                                Style.EMPTY
+                                                        .withItalic(false)
+                                                        .withColor(Formatting.GRAY)
+                                        ).append(
+                                                Text.literal(owner).setStyle(
+                                                        Style.EMPTY
+                                                                .withItalic(false)
+                                                                .withColor(Formatting.YELLOW)
+                                                )
+                                        )
+                        )
+                )
+        );
+
+        // Members info
+        int maxSize = FactionsMod.CONFIG.MAX_FACTION_SIZE;
+        boolean isMaxSize = FactionsMod.CONFIG.MAX_FACTION_SIZE > -1;
+
+        List<Text> membersLore = new java.util.ArrayList<>(members.stream()
+                .map(u -> Text.literal(cache.getByUuid(u.getID())
+                                .orElse(new GameProfile(Util.NIL_UUID, "{Uncached Player}"))
```

Comment:
"{Unknown Username}" makes more sense to me then "{Uncached Player}"

Reply:
Changed to `{Unknown Player}` for sake of consistency with [MemberGui.java#L65](https://github.com/ickerio/factions/blob/2b2ac2fddba20b060ee52ab63b88f043d0dd5d27/src/main/java/io/icker/factions/ui/MemberGui.java#L65). Minecraft disconnects you if that username is over 16 characters.